zappa/utilities.py

```diff
@@ -592,3 +599,10 @@ def merge_headers(event):
     for h in multi_headers.keys():
         multi_headers[h] = ", ".join(multi_headers[h])
     return multi_headers
+
+
+def validate_json_serializable(function_args: dict) -> None:
```

Comment:
I might suggest we change the signature of this utility function such that it's more generic and reusable elsewhere if needed.  Other Python objects besides dicts can be serialized into valid JSON-encoded strings.

The simplest way to accomplish this would probably be to allow this function accept any amount of objects (in arg and/or kwarg form) and check them all in one go by creating a simple tuple out of them to pass to `json.dumps`.

```python
def validate_json_serializable(*args: Any, **kwargs: Any) -> None:
    try:
         json.dumps((args, kwargs))
     except ...
```

This also removes the burden on the function caller to convert what are potentially multiple objects into a single data structure beforehand, which, for example, would also allow us to trim a line above where it is being used where we are doing exactly that:

```python
function_args = {"args": args, "kwargs": kwargs}
validate_json_serializable(function_args)
```

becomes:

```python
validate_json_serializable(*args, **kwargs)
```

Alternatively, we could just broaden the function definition and leave all else as-is:

```python
def validate_json_serializable(obj: Any) -> None:
    try:
        json.dumps(obj)
    except ...
```

I slightly prefer the former as it doesn't require the user to have to potentially convert multiple objects into a single data structure before using the function (like we are having to do now, by converting args+kwargs to a dict before passing to this function).  WDYT?

Reply:
My focus was this single use case, but that's a good point.  If we include it in `utilities.py` it should support more wide usage.

Updated!